src/app/components/mno-product-selector/mno-product-selector.controller.coffee

```diff
@@ -66,9 +71,39 @@
 
       promise.then(
         (response) ->
+          # Extract the categories
+          categories = []
+          response.data.forEach((p) ->
+            categories = categories.concat(p.categories))
+          $ctrl.categories = _.uniq(categories)
```

Comment:
Use `map` to map an array rather than iterating and pushing
```coffeescript
categories  = response.data.map (product) -> product.categories
```

However, we can also use Coffescript array comprehension, which will compiles to a `for` loop which is faster than `map` and `forEach`:
```coffeescript
categories = (product.categories for categories in response.data)
```